pkg/api/rest/update.go

```diff
@@ -85,6 +85,12 @@ func BeforeUpdate(strategy RESTUpdateStrategy, ctx api.Context, obj, old runtime
 	} else {
 		objectMeta.Namespace = api.NamespaceNone
 	}
+	// Ensure requests cannot update generation
+	oldMeta, err := api.ObjectMetaFor(old)
+	if kerr != nil {
+		return kerr
+	}
+	objectMeta.Generation = oldMeta.Generation
```

Comment:
ds: https://github.com/kubernetes/kubernetes/blob/085b17239503363f0fb4b0c960bf185b49495261/pkg/registry/daemonset/strategy.go#L75
rc: https://github.com/kubernetes/kubernetes/blob/085b17239503363f0fb4b0c960bf185b49495261/pkg/registry/controller/strategy.go#L73
rs: https://github.com/kubernetes/kubernetes/blob/085b17239503363f0fb4b0c960bf185b49495261/pkg/registry/replicaset/strategy.go#L74
ingress: https://github.com/kubernetes/kubernetes/blob/085b17239503363f0fb4b0c960bf185b49495261/pkg/registry/ingress/strategy.go#L67
petset: https://github.com/kubernetes/kubernetes/blob/085b17239503363f0fb4b0c960bf185b49495261/pkg/registry/petset/strategy.go#L67
deployments: https://github.com/kubernetes/kubernetes/blob/085b17239503363f0fb4b0c960bf185b49495261/pkg/registry/deployment/strategy.go#L81

Not sure there is another resource using it anywhere else. Seems like an established pattern.